.github/workflows/nightly.yml

```diff
@@ -0,0 +1,32 @@
+# Builds the proxy on the nightly toolchain to help catch Rust regressions before they bite us hit
+# stable.
+name: nightly
```

Comment:
perhaps this should be named "nightly rust" or something, so it doesn't imply that it's a nightly build of the _proxy_? not a hard blocker